pyccel/ast/builtin_objects/list_functions.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+The List container has a number of built-in methods that are 
+always available.
+
+This module contains objects which describe these methods within Pyccel's AST.
+"""
+
+from pyccel.ast.datatypes import NativeVoid, NativeGeneric, NativeHomogeneousList
+from pyccel.ast.internals import PyccelInternalFunction
+
+
+class ListAppend(PyccelInternalFunction):
+    """
+    Represents a call to the .append() method.
+
+    Represents a call to the .append() method of an object with a list type,
+    which adds an element to the end of the list. This method returns `None`.
+    The append method is called as follows:
+
+    >>> a = [1]
+    >>> a.append(2)
+    >>> print(a)
+    [1, 2]
+
+    Parameters
+    ----------
+    list_variable : Variable
+        The variable representing the list.
+    
+    new_elem : Variable
+        The argument passed to append() method.
+    """
+    __slots__ = ("_list_variable", "_append_arg")
+    _attribute_nodes = ("_list_variable", "_append_arg")
+    _dtype = NativeVoid()
+    _shape = None
+    _order = None
+    _rank = 0
+    _precision = -1
+    _class_type = NativeHomogeneousList()
+    name = 'append'
+
+    def __init__(self, list_variable, new_elem) -> None:
+        conditions = (
+            new_elem.dtype is not NativeGeneric() and
+            list_variable.dtype == new_elem.dtype and
+            list_variable.precision == new_elem.precision and
+            list_variable.rank - 1 == new_elem.rank
+        )
+        is_homogeneous = list_variable.dtype is not NativeGeneric() and conditions
```

Comment:
Why do we need the `conditions` boolean variable? Could we not define `is_homogeneous` directly?
```suggestion
        is_homogeneous = (
            new_elem.dtype is not NativeGeneric() and
            list_variable.dtype is not NativeGeneric() and
            list_variable.dtype == new_elem.dtype and
            list_variable.precision == new_elem.precision and
            list_variable.rank - 1 == new_elem.rank
        )
```

Reply:
Yes it's possible, I'll refactor it in my next commit

Reply:
@Farouk-Echaref please mention if and where my comment was addressed.

Reply:
@yguclu fixed

Reply:
Where?

Reply:
commit `ce7ec14dd05f8adbe1f680325a175a173733837d`
![commit](https://github.com/pyccel/pyccel/assets/66710845/17d51fd1-7d9e-4841-b743-89c37cc46ee4)

Reply:
Thanks! no need for the screenshot 😉